### ISSUE TITLE
Please allow not to detect the backup GPT table by default

### ISSUE THREAD

**Issue by funh2029** [Quyzi/gpt#89]
My program run on an embedded device. When I execute the following statement, I get an error "invalid GPT signature":

```let disk = gpt::GptConfig::new().writable(false).open(path)?;```

Through tracing, I found that the execution flow is as follows:
```
open->open_from_device->header::read_backup_header->file_read_header:
    if sigstr != "EFI PART" {
        return Err(Error::new(ErrorKind::Other, "invalid GPT signature"));
    };
```
Obviously, the embedded device only has the primary partition table, but not the backup table. However, an error is reported due to the detection of the backup table.

**Comment by soerenmeier** [comment]
Can you try the following branch to see if it works?
https://github.com/soerenmeier/gpt/tree/allow_faulty_header

**Comment by funh2029** [comment]
> Can you try the following branch to see if it works? https://github.com/soerenmeier/gpt/tree/allow_faulty_header

no, it doesn't work.

**Comment by funh2029** [comment]
Just add an unwrap_or.

**Comment by funh2029** [comment]
> Can you try the following branch to see if it works? https://github.com/soerenmeier/gpt/tree/allow_faulty_header

Yes, it's work.

**Comment by funh2029** [comment]
Or just like this:

```
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -157,7 +157,7 @@ impl GptConfig {

         // Proper GPT disk, fully inspect its layout.
         let h1 = header::read_primary_header(&mut device, self.lb_size)?;
-        let h2 = header::read_backup_header(&mut device, self.lb_size)?;
+        let h2 = header::read_backup_header(&mut device, self.lb_size).unwrap_or(h1.clone());
         let table = partition::file_read_partitions(&mut device, &h1, self.lb_size)?;
         let disk = GptDisk {
             config: self,
```

**Comment by soerenmeier** [comment]
I've published v4.0.0-rc.1 which has this behaviour by default.
https://docs.rs/gpt/4.0.0-rc.1/gpt/index.html 4.0.0 will be following soon

**Comment by soerenmeier** [comment]
This is now the default in v4.0.0
